RecoEgamma/PhotonIdentification/src/PhotonIsolationCalculator.cc

```diff
@@ -661,19 +664,19 @@ double PhotonIsolationCalculator::calculateHcalTowerIso(const reco::Photon* phot
 {
 
   edm::Handle<CaloTowerCollection> hcalhitsCollH;
- 
-  iEvent.getByToken(hcalCollection_, hcalhitsCollH);
-  
-  const CaloTowerCollection *toww = hcalhitsCollH.product();
-
+  const CaloTowerCollection *toww = nullptr; 
+  if (not hcalCollection_.isUninitialized()){
+    iEvent.getByToken(hcalCollection_, hcalhitsCollH);
+    toww = hcalhitsCollH.product();
+  }
   double hcalIsol=0.;
   
   //std::cout << "before iso call" << std::endl;
   EgammaTowerIsolation phoIso(RCone,
                               RConeInner,
                               eMin,depth,
                               toww);
-  hcalIsol = phoIso.getTowerEtSum(photon);
+  hcalIsol = (toww != nullptr ? phoIso.getTowerEtSum(photon) : 0.);
```

Comment:
as above: better move the computation behind the ```if (not hcalCollection_.isUninitialized()){``` to avoid unnecesary initialization/instantiation of the phoIso